File: perl/87_query_board.pl
open my $fh, '<', $ARGV[0];

my $board = [];
for ( my $i = 0; $i != 256; $i++ ) {
	push @$board, [];

	for ( my $j = 0; $j != 256; $j++ ) {
		push @{ $board->[0] }, 0;
	}
}

for my $line ( <$fh> ) {
	$line =~ s/[\n\r]+//g;
	next if $line eq '';

	(my $cmd, my @args) = split / /, $line;

	if ( $cmd eq 'SetCol' ) {
		my $col = $args[0];
		my $x = $args[1];

		for ( my $i = 0; $i != 256; $i++ ) {
			$board->[$i]->[$col] = $x;
		}
	}
	elsif ( $cmd eq 'SetRow' ) {
		my $row = $args[0];
		my $x = $args[1];

		for ( my $i = 0; $i != 256; $i++ ) {
			$board->[$row]->[$i] = $x;
		}
	}
	elsif ( $cmd eq 'QueryCol' ) {
		my $col = $args[0];

		my $sum = 0;
		for ( my $i = 0; $i != 256; $i++ ) {
			$sum += $board->[$i]->[$col];
		}

		print "$sum\n";
	}
	elsif ( $cmd eq 'QueryRow' ) {
		my $row = $args[0];

		my $sum = 0;
		for ( my $i = 0; $i != 256; $i++ ) {
			$sum += $board->[$row]->[$i];
		}

		print "$sum\n";
	}
}

close $fh;

exit 0;

File: python/87_query_board.py
import sys

board = [[0 for j in range(256)] for i in range(256)]

with open(sys.argv[1], 'r') as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        (cmd, args) = line.split(' ', 1)
        args = args.split(' ')

        if cmd == 'SetCol':
            col = int(args[0])
            x = int(args[1])

            for i in range(256):
                board[i][col] = x
        elif cmd == 'SetRow':
            row = int(args[0])
            x = int(args[1])

            for i in range(256):
                board[row][i] = x
        elif cmd == 'QueryCol':
            col = int(args[0])

            print sum([r[col] for r in board])
        elif cmd == 'QueryRow':
            row = int(args[0])

            print sum(board[row])

